frontend-public/src/components/dashboard/mine/variances/VarianceTable.js

```diff
@@ -15,144 +15,208 @@ const propTypes = {
-                  key={file.mine_document_guid}
-                  onClick={() => downloadFileFromDocumentManager(file.document_manager_guid)}
-                  // Accessibility: Event listener
-                  onKeyPress={() => downloadFileFromDocumentManager(file.document_manager_guid)}
-                  // Accessibility: Focusable element
-                  tabIndex="0"
-                >
-                  {file.document_name}
-                </a>
-              </div>
-            ))
-          : Strings.EMPTY_FIELD}
-      </div>
-    ),
-  },
-];
+  handleConditionalEdit = (code) => code === Strings.VARIANCE_APPLICATION_CODE;
+
+  transformRowData = (variances, codeHash, statusHash) =>
+    variances.map((variance) => ({
+      key: variance.variance_guid,
+      variance,
+      isEditable: this.handleConditionalEdit(variance.variance_application_status_code),
+      status: statusHash[variance.variance_application_status_code],
+      compliance_article_id: codeHash[variance.compliance_article_id] || Strings.EMPTY_FIELD,
+      expiry_date: formatDate(variance.expiry_date) || Strings.EMPTY_FIELD,
+      issue_date: formatDate(variance.issue_date) || Strings.EMPTY_FIELD,
+      note: variance.note,
+      received_date: formatDate(variance.received_date) || Strings.EMPTY_FIELD,
+      isOverdue: variance.expiry_date && Date.parse(variance.expiry_date) < new Date(),
+      documents: variance.documents,
+    }));
+
+  handleOpenModal = (event, isEditable, variance) => {
+    event.preventDefault();
+    if (isEditable) {
+      this.props.openEditVarianceModal(variance);
+    } else {
+      this.props.openViewVarianceModal(variance);
+    }
+  };
 
-export const VarianceTable = (props) => (
-  <div>
-    <Table
-      align="left"
-      pagination={false}
-      columns={columns(props.isApplication)}
-      locale={{
-        emptyText: props.isApplication
-          ? "No pending variance applications"
-          : "No approved variances",
-      }}
-      dataSource={transformRowData(
-        props.variances,
-        props.complianceCodesHash,
-        props.varianceStatusOptionsHash
-      )}
-    />
-  </div>
-);
+  columns = (isApplication) => [
+    {
+      title: "",
+      dataIndex: "isOverdue",
+      width: 10,
+      render: (isOverdue) => (
+        <div title="">
+          {isOverdue ? <img className="padding-small" src={RED_CLOCK} alt="expired" /> : ""}
+        </div>
+      ),
+    },
+    {
+      title: "Code Section",
+      dataIndex: "compliance_article_id",
+      render: (text, record) => (
+        <div title="Code Section" style={this.errorStyle(record.isOverdue)}>
+          {text}
+        </div>
+      ),
+    },
+    {
+      title: "Submission Date",
+      dataIndex: "received_date",
+      // className on the column - applies to the entire column when full size
+      //  This attribute has no effect on the responsive view, same class is added to the div to handle responsive views
```

Comment:
Could you wrap this long comment line? 😃

Reply:
done!